airbyte-webapp/src/hooks/services/FormChangeTracker/hooks.ts

```diff
@@ -22,18 +24,18 @@ export const useFormChangeTrackerService = (): FormChangeTrackerServiceApi => {
 
   const clearFormChange = useCallback(
     (id: string) => {
-      setChangedFormsById({ ...changedFormsById, [id]: false });
+      setChangedFormsById({ ...changedFormsByIdRef.current, [id]: false });
     },
-    [changedFormsById, setChangedFormsById]
+    [changedFormsByIdRef, setChangedFormsById]
   );
 
   const trackFormChange = useCallback(
     (id: string, changed: boolean) => {
-      if (Boolean(changedFormsById?.[id]) !== changed) {
-        setChangedFormsById({ ...changedFormsById, [id]: changed });
+      if (Boolean(changedFormsByIdRef.current?.[id]) !== changed) {
+        setChangedFormsById({ ...changedFormsByIdRef.current, [id]: changed });
       }
     },
-    [changedFormsById, setChangedFormsById]
+    [changedFormsByIdRef, setChangedFormsById]
```

Comment:
I feel like this change might be excessive? But not sure. If we need to store the value in a ref maybe the `useChangedFormsById` should be a ref instead?

Reply:
It's important because otherwise it's easy to run into endless update cycles (you have an effect that needs the reference of the latest trackFormChange, which updates the changedFormsById, which updates the reference of trackFormChange, which re-triggers the effect and so on)

changedFormsById could probably be a ref, but it seems like it's important that it's global state and there is no createGlobalRef, so that seems like the cleanest solution to me.

Happy to implement another path if you want though.

Reply:
I don't want to hold up this PR and if this fixes it I'm fine with it.
I'd need to dig into the code again to see what I can come up with. Perhaps all this could be streamlined but would require investigation.

Reply:
It would be nice if we would never actually need multi-form tracking, that would allow us to throw away most of this code. it felt out of scope for this PR though.